tests/util/test_dict_path.py

```diff
@@ -88,6 +88,18 @@ def test_add() -> None:
         assert (ComposedPath(path_str=first) + ComposedPath(path_str=second)).to_str() == first + "." + second
 
 
+def test_in() -> None:
+    items = [
+        ("a.b.c", "a.b", True),
+        ("a.c[z=www].d.e", "c.a", False),
+        ("a.c[z=www].d.g", "a.c.d", True),
```

Comment:
I think this should be false, those two dicts path are not equivalent, the first one doesn't lead to an object that is in the object pointed to by the second one